runtime/src/qio/qio_popen.c

```diff
@@ -384,16 +384,26 @@ qioerr qio_proc_communicate(
 
   if( threadsafe ) {
     // lock all three channels.
+    // but unlock them immediately and set them to NULL
+    // if they are already closed.
```

Comment:
Won't unlocking cause a potential race?  If two tasks are attempting to lock a closed channel and this one gets to lock it first, then the unlock would cause a race between setting input to NULL and locking it, right?  Can we be assured that no one else will try to lock a closed channel?